lib/src/summarize.dart

```diff
@@ -4,12 +4,95 @@
 
 library dart_pad.summarize;
 
-/// Instances of this class take string input of dart code and output a text 
-/// description ofthe code's size, packages, and other useful information
+import '../services/dartservices.dart';
+/// Instances of this class take string input of dart code as well as an analysis result,
+/// and output a text description ofthe code's size, packages, and other useful information.
 class Summarizer {
   
-  String summarize(String input) {
-    return "foo";
+  SummarizeToken storage;
+
+  bool resultsPresent;
+  
+  Summarizer(String input, [AnalysisResults analysis]) {
+    if (input == null) throw new ArgumentError("Input can't be null.");
+    resultsPresent = !(analysis == null);
+    storage = new SummarizeToken(input, analysis);
   }
+  
+  String returnAsSimpleSummary() {
+    String summary = 'Summary (Under Development):';
+    return summary;
+  }
+  
+  String returnAsGistMarkDown() {
+    if (resultsPresent) {
+      String summary = '<pre><code><b>Summary (Under Development)</b><p/>';
```

Comment:
Markdown is very simple markup lanuage. It's doesn't use html tags. It's what dart's dartdoc syntax is based on. Here's a quick overview: https://help.github.com/articles/markdown-basics/

I would go very light w/ the markdown here for now. Probably just make sure to do `\n\n` at the end of every string, so it forces a new line.